_pyinstaller_hooks_contrib/stdhooks/hook-niquests.py

```diff
@@ -0,0 +1,26 @@
+# ------------------------------------------------------------------
+# Copyright (c) 2023 PyInstaller Development Team.
+#
+# This file is distributed under the terms of the GNU General Public
+# License (version 2.0 or later).
+#
+# The full license is available in LICENSE, distributed with
+# this software.
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+# ------------------------------------------------------------------
+
+from PyInstaller.utils.hooks import collect_submodules, is_package
+
+
+hiddenimports = [
+    *collect_submodules("niquests"),
+    *collect_submodules("urllib3"),
+    *collect_submodules("urllib3_future"),
+    *collect_submodules("jh2"),
+    *collect_submodules("h11"),
```

Comment:
Are these really all required? The provided test seems to be working (at least on linux) with just `urllib3.contrib.resolver.system` and `niquests.packages` added to hidden imports. And even the latter is needed only because of [these imports](https://github.com/jawah/niquests/blob/f42b2bb873e3e516ad8044d69a5c72a4f8b8c88e/src/niquests/__init__.py#L49-L55) that imply that `niquests.packages` should be a package whereas it is a module, and throw off our analysis.

Reply:
It is necessary. We can't be really precise here.

I have added complementary tests. But the main idea here, is that "urllib3-future" have many lazy loaded sub modules. not just `urllib3.contrib.resolver.system` but  `urllib3.contrib.hface.protocols.http3` or even `urllib3.contrib.webextensions`. There's plan for others in the future, so the hook is more resilient this way.
FYI: I've removed h11, jh2 and qh3 explicit collection as PyInstaller seems to find it automatically.

Regards,

Reply:
Hmm, the extended test code still seems to run for me with just the afore-mentioned two hidden imports, but I get what you are saying.

Just to clarify, though, are these lazy imports done by `niquest` or are they coming from within `urllib3` / `urllib3_future` itself?

As for the PR,
1. add `niquest` to [`requirements-test-libraries.txt`](https://github.com/pyinstaller/pyinstaller-hooks-contrib/blob/master/requirements-test-libraries.txt) so that the test will be ran when the package is updated
2. add a [news fragment](https://github.com/pyinstaller/pyinstaller-hooks-contrib/blob/master/news/README.txt) for the changelog - the entry can be simply `Add hook for ``niquests``.`

Reply:
both niquests and urllib3/urllib3-future are concerned.

1. done.
2. already added, maybe you missed it.